hnn_core/parallel_backends.py

```diff
@@ -0,0 +1,296 @@
+"""Parallel backends"""
+
+# Authors: Blake Caldwell <blake_caldwell@brown.edu>
+#          Mainak Jas <mainakjas@gmail.com>
+
+import os
+import sys
+import multiprocessing
+import shlex
+import pickle
+import codecs
+from warnings import warn
+from subprocess import Popen, PIPE
+
+_BACKEND = None
+
+
+def _gather_trial_data(sim_data, net, n_trials):
+    """Arrange data by trial
+
+    To be called after simulate(). Returns list of Dipoles, one for each trial,
+    and saves spiking info in net (instance of Network).
+    """
+    dpls = []
+
+    for idx in range(n_trials):
+        dpls.append(sim_data[idx][0])
+        spikedata = sim_data[idx][1]
+        net.spikes._times.append(spikedata[0])
+        net.spikes._gids.append(spikedata[1])
+        net.gid_dict = spikedata[2]  # only have one gid_dict
+        net.spikes.update_types(net.gid_dict)
+
+    return dpls
+
+
+class JoblibBackend(object):
+    """The JoblibBackend class.
+
+    Parameters
+    ----------
+    n_jobs : int | None
+        The number of jobs to start in parallel. If None, then 1 trial will be
+        started without parallelism
+
+    Attributes
+    ----------
+    n_jobs : int
+        The number of jobs to start in parallel
+    """
+    def __init__(self, n_jobs=1):
+        self.type = 'joblib'
```

Comment:
This attribute needs to be documented @blakecaldwell with reference to my comment https://github.com/jonescompneurolab/hnn-core/pull/79/files#r454614746. Sorry if that was too ambiguous

Reply:
Ah! I didn't see it. In fact, that attribute wasn't even necessary. Removed.